src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -992,6 +992,61 @@ class VariantSampleMatrix[T](val metadata: VariantMetadata,
         "sa" -> saSignature),
       Array("s"))
   }
+
+  def annotateVariantsKeyTable(kt: KeyTable, code: String) = {
+    val ktKeyTypes = kt.keySignature.fields.map(_.`type`)
+
+    if (ktKeyTypes.size != 1 || ktKeyTypes(0) != TVariant)
+      fatal(s"Key signature of KeyTable must be 1 field with type `Variant'. Found `${kt.keySignature}'")
+
+    val ktSig = kt.signature
+
+    val inserterEc = EvalContext(Map("va" -> (0, vaSignature), "table" -> (1, ktSig)))
+
+    val (finalType, inserter) =
+      buildInserter(code, vaSignature, inserterEc, Annotation.VARIANT_HEAD)
+
+    val keyedRDD = kt.rdd.map { case (k: Row, v) => (k(0).asInstanceOf[Variant], kt.mergeKeyAndValue(k, v)) }
+
+    val ordRdd = OrderedRDD(keyedRDD, None, None)
+
+    annotateVariants(ordRdd, finalType, inserter)
+  }
+
+  def annotateVariantsKeyTable(kt: KeyTable, vdsKey: String, code: String) = {
+
+    val vdsKeyEc = EvalContext(Map("v" -> (0, TVariant), "va" -> (1, vaSignature)))
+
+    val (vdsKeyType, vdsKeyF) = Parser.parseExpr(vdsKey, vdsKeyEc)
+
+    if (kt.keySignature != vdsKeyType)
+      fatal(s"Key signature of KeyTable, `${kt.keySignature}', must match type of computed key, `${vdsKeyType}'.")
+
+    val ktSig = kt.signature
+
+    val inserterEc = EvalContext(Map("va" -> (0, vaSignature), "table" -> (1, ktSig)))
+
+    val (finalType, inserter) =
+      buildInserter(code, vaSignature, inserterEc, Annotation.VARIANT_HEAD)
+
+    val ktRdd = kt.rdd.map { case (k, v) => (k, kt.mergeKeyAndValue(k, v)) }
```

Comment:
You might not want to include this now, but I think KeyTable should have an RDD[Annotation] which should have values of the entire schema, and then (I think) we only need a function that slaps the keys on, to give an RDD[(Annotation, Annotation)] of values (key schema, entire schema).  In particular, I think we shouldn't ever pull out the non-keys value part.

I already mentioned this to @jigold although she's working on other stuff so I don't think she did anything yet.